fedimint-server/src/net/peers.rs

```diff
@@ -78,6 +80,52 @@ struct PeerConnectionStateMachine<M> {
     state: PeerConnectionState<M>,
 }
 
+struct PeerStatusQuery {
+    response: oneshot::Sender<PeerStatus>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum PeerStatus {
+    Connected,
+    Disconnected,
+}
+
+type PeerStatusChannelSender = Sender<PeerStatusQuery>;
+type PeerStatusChannelReceiver = Receiver<PeerStatusQuery>;
+
+#[derive(Clone)]
+pub struct PeerStatusClient(HashMap<PeerId, PeerStatusChannelSender>);
+
+impl PeerStatusClient {
+    pub async fn query_all_status(&self) -> HashMap<PeerId, anyhow::Result<PeerStatus>> {
+        let mut results = HashMap::new();
+        for (peer_id, sender) in self.0.iter() {
+            let (response_sender, response_receiver) = oneshot::channel();
+            let query = PeerStatusQuery {
+                response: response_sender,
```

Comment:
`response_sender` or `response_tx` or something is better name than `response`, especially in a `struct`, IMO.